aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -500,6 +515,48 @@ Use <xref:Microsoft.AspNetCore.Components.NavigationManager> to manage URIs and
 
 :::moniker-end
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Enhanced navigation and form handling
+
+*This section applies to Blazor Web Apps.*
+
+Blazor Web Apps are capable of two types of routing for page navigation and form handling requests:
+
+* Normal navigation: For all cross-document/cross-origin navigation, a full-page reload is triggered for the request URL.
+* Enhanced navigation: For some same-document/same-origin navigation&dagger;, Blazor intercepts the request and performs a `fetch` request instead. Blazor then patches the response content into the page's DOM. Blazor's enhanced navigation and form handling avoid the need for a full-page reload and preserves more of the page state, so pages load faster, usually without losing the user's scroll position on the page.
```

Comment:
Just to clarify - cross-document and cross-origin are two different things. Cross-document means that you lose JavaScript/Document state between navigations. Cross-origin means that the navigation results in the origin changing. A navigation can be same-origin but still cross-document. Hitting the refresh button is an example of this.

Cross-origin navigations are always cross-document. Same-origin navigations have the _potential_ to be same-document. Enhanced navigation is an example of a same-origin and same-document navigation. If enhanced navigation is explicitly disabled (and there's no interactive router), then most same-origin navigations will be cross-document (link clicks, hitting the back/forward button, etc.).

It's probably not necessary to elaborate on all this in these docs, but maybe we could phrase these definitions like:
```suggestion
* Normal navigation: Cross-document navigation; a full-page reload is triggered for the request URL.
* Enhanced navigation: Same-document navigation;&dagger;, Blazor intercepts the request and performs a `fetch` request instead. Blazor then patches the response content into the page's DOM. Blazor's enhanced navigation and form handling avoid the need for a full-page reload and preserves more of the page state, so pages load faster, usually without losing the user's scroll position on the page.
```

I think the key is to be clear that the same-document/cross-document decision doesn't happen _before_ Blazor decides whether to do an enhanced or normal navigation. The same-origin/cross-origin decision happens first (by nature of the URL being navigated to). Then, if the navigation is same-origin, Blazor decides whether to perform a same-document or cross-document navigation.

Apologies if it seems like I'm explaining this to death, or if this is information you already knew!